3401_project.ipynb

```diff
@@ -1027,6 +1027,681 @@
+      "text/plain": [
+       "<Figure size 640x480 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "songs.hist(column=\"popularity\", bins=10)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3.4 Look at structure of artist_name and genre"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "artist_name\n",
+       "Wolfgang Amadeus Mozart    377\n",
+       "Zarobi                     257\n",
+       "Shakuhachi Sakano          197\n",
+       "Ludwig van Beethoven       149\n",
+       "Johann Sebastian Bach      139\n",
+       "                          ... \n",
+       "Doctor P                     1\n",
+       "STUCA                        1\n",
+       "Nitepunk                     1\n",
+       "Supereffective               1\n",
+       "Barış Manço                  1\n",
+       "Name: count, Length: 11048, dtype: int64"
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "songs[\"artist_name\"].value_counts()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "genre\n",
+       "black-metal       889\n",
+       "gospel            845\n",
+       "forro             836\n",
+       "emo               824\n",
+       "indian            814\n",
+       "                 ... \n",
+       "grindcore          62\n",
+       "chicago-house      52\n",
+       "show-tunes         50\n",
+       "detroit-techno     43\n",
+       "romance            16\n",
+       "Name: count, Length: 81, dtype: int64"
+      ]
+     },
+     "execution_count": 12,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "songs[\"genre\"].value_counts()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 4 - Preprocessing"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4.1 Drop ID columns\n",
+    "\n",
+    "There are several columns in the dataset that are unique to each example/track (index, track_name, track_id). These need to be removed so that the models do not acc
```

Comment:
I think year should be dropped as well.
I am not sure that artist_name should be remained or not. we need to discuss about this.

Reply:
Agreed. 

For artist_name, I think we'll need to drop it for now because it breaks the one-hot encoder. I've leave in a TODO to remind us in case we come up with something clever